src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -325,8 +333,16 @@ private static bool MyBreakHandler(ConsoleControl.ConsoleBreakSignal signal)
             switch (signal)
             {
                 case ConsoleControl.ConsoleBreakSignal.CtrlBreak:
-                    // Break into script debugger.
-                    BreakIntoDebugger();
+                    if (s_cpp.NonInteractive)
+                    {
+                        //ControlBreak mimics ControlC in Noninteractive shells
+                        SpinUpBreakHandlerThread(false);
```

Comment:
@PaulHigin because I launch powershell in its own process group, ctrl+C is implicitly disabled for the lauched process and its children (see [Remarks section](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682425(v=vs.85).aspx) of `CreateProcess`). This allows parent applications like my own to better control the propogation of `SIGINT` signals and prevent ctrl+c from broadcasting to all processes in the console.

I'll change `shouldEndSession` to `true`. I was blindly looking to imitate ctrl+c but `true` does make more sense.  Thanks for the above snippet of the former behavior. I was looking for the state before entering the debugger yesterday and couldn't find it in the history.

I think sticking with `SpinUpBreakHandlerThread` is likely the best pattern to follow now.